src/anemoi/inference/inputs/gribfile.py

```diff
@@ -27,11 +27,12 @@ class GribFileInput(GribInput):
     trace_name = "grib file"
 
     def __init__(self, context, path, *, namer=None, **kwargs):
-        super().__init__(context, namer=namer, **kwargs)
+        super().__init__(context, namer=namer)
         self.path = path
+        self.kwargs = kwargs
```

Comment:
Using the kwargs like this is dangerous. For now this works because the super class doesn't use them, but if in the future the super class starts accepting any kwargs it will break.

Why don't we make `mask` part of the constructor of `EkdInput`, as an optional parameter? You then store it in a `self.mask` and pass it through where needed. It should be compatible with all other ekd-derived inputs.

Reply:
I saw that extra kwargs were passed to `_create_input_state` in the `EkdInput` instead of to the constructor so I  did the same. I agree it's not ideal. I'll edit this, thanks!